docs/Components/SudokuBoard/BoardLogic.md

```diff
@@ -0,0 +1,46 @@
+# Classic Variant
+
+## Insertion Logic
+
+```mermaid
```

Comment:
Awesome use of a mermaid chart for documentation! I do have a nit: would be nice if you used [this format](https://mermaid.js.org/syntax/flowchart.html#graph-declarations-with-spaces-between-vertices-and-link-and-without-semicolon) of having a diamond shape connected to the User Selects Cell which says something about the state of that pencil button (not sure if you have an official term for it in docs already?) and then the lines to the next two parts have text on them declaring the state of the pencil that causes each path to be taken.

Reply:
Do you mean something like this?

```mermaid
flowchart
    PencilButtonState{{Note Button State}}
    InsertStart[User Selects Cell]
    InsertValue[User Presses HotKey/Button <br> to insert value into Cell]
    InsertNote[User Presses HotKey/Button <br>to insert note into Cell]
    PushHistory[Board pushes existing cell state <br> to ActionHistory array]
    UpdateCell[Cell is updated with new value]

    PencilButtonState --> InsertStart
    InsertStart --> |If Note button is disabled|InsertValue
    InsertStart --> |If Note button is enabled|InsertNote
    InsertValue --> |If Insert is Valid|PushHistory
    InsertNote --> |If Insert is Valid|PushHistory
    PushHistory --> UpdateCell
```

Reply:
Almost, basically just flip the Note Button State and the User Select Cell (seems to be a standard style to use diamond shape to do decision/if logic)
![image](https://github.com/Sudokuru/Frontend/assets/89822791/2b4fc93b-5c01-4f84-80ff-b046b7e633f0)

Reply:
I made this change, and also updated the Undo logic diagram to have the diamond shape as well for the "User enters undo" decision.